src/screen.cpp

```diff
@@ -734,7 +734,19 @@ static void s_update(screen_t *scr, const wcstring &left_prompt, const wcstring
     // Output the left prompt if it has changed.
     if (left_prompt != scr->actual_left_prompt) {
         s_move(scr, 0, 0);
-        s_write_str(scr, left_prompt.c_str());
+        if (clr_eol) {
+            s_write_mbs(scr, clr_eol);
+        }
```

Comment:
I would move this to after the last write to minimize flashes of contents/repaints. But first - in what case is it required, given that it would presumably apply to even single-line prompts but we don't have any problems with those now, even without it?

Reply:
That's a good idea. I will make that change. It might make the logic slightly more complicated, since currently it goes "clr, line+terminator", which will need to be changed into "terminator+line, clr". Just some indexing stuff to take care of :)

I think the fact that it's being printed on a single line prompt is also just an artefact of that, so I can definitely eliminate it. We don't have a problem for single line prompts because rendering the command also clears to the end of the line.

Reply:
Updated, let me know if you have any suggestions to make the logic clearer.